walkeeper/src/wal_storage.rs

```diff
@@ -339,69 +383,67 @@ impl Storage for PhysicalStorage {
             }
         }
 
+        Ok(())
+    }
+
+    fn flush_wal(&mut self) -> Result<()> {
+        if self.flush_record_lsn == self.write_record_lsn {
+            // no need to do extra flush
+            return Ok(());
+        }
+
+        if let Some(mut unflushed_file) = self.file.take() {
+            self.fdatasync_file(&mut unflushed_file)?;
```

Comment:
Option can be matched with `ref` keyword to its contents to avoid taking/returning it around.

Reply:
I need to call `self.fdatasync_file` in this if, so with `ref` borrow I get an error:
```
cannot borrow `*self` as immutable because it is also borrowed as mutable
immutable borrow occurs hererustc[E0502](https://doc.rust-lang.org/error-index.html#E0502)
```